docs/proposals/apiman-integration.md

```diff
@@ -0,0 +1,135 @@
+# APIMan Integration
+
+## Abstract
+This proposes a design for integrating [APIMan](http://www.apiman.io/) with OpenShift in order to provide  micro-services governance for API service providers.
+
+## Motivation
+Providers of services may have the need to control who and how their services are consumed.  This will be realized as policies defined by providers to control their service in any number of ways (e.g security, throttling/quota, billing and metrics)<sup>[1](#r1)</sup>. 
+
+## Constraints and Assumptions
+### Authentication
+The OAuth token identifying a user in OpenShift will be utilized by APIMan to associate service policy to their identity.  This will ensure a user's identity is ubiquitous across the cluster when interacting with OpenShift and the APIMan management console.
+
+### Deployment
+#### Communication
+APIMan components (e.g. gateway, management interface) integrated with the cluster will utilize mutual TLS for internal communication.
+
+#### Storage
+APIMan is capable of using several backends (e.g. ElasticSearch, Postgresql) to store configuration, policy, and metrics.  The initial integration will utilize an ElasticSearch cluster dedicated to APIMan.  The ElasticSearch image will be shared with aggregated logging.  The [ACL plugin](https://github.com/fabric8io/openshift-elasticsearch-plugin)<sup>[2](#r2)</sup> that is deployed as part of aggregated logging will be disabled in the APIMan ElasticSearch deployment. Additionally, the ElasicSearch index management<sup>[3](#r3)</sup>  will be modified so that it will not cull APIMan data.
```

Comment:
APIMan/logging may be able to share an ES base image with all of the plugins imported. The configurations are going to be different, so that still implies different images, unless we want to mount config over at runtime? That wouldn't be too hard, just... different.

Don't think we'll modify curator, just configure it to leave the APIMan data alone.